gitbutler-ui/src/lib/components/BranchLane.svelte

```diff
@@ -17,11 +27,37 @@
 
 	export let branch: Branch;
 	export let isUnapplied = false;
-	export let branchCount = 1;
 
-	$: selectedOwnership = writable(Ownership.fromBranch(branch));
+	const selectedOwnership = Ownership.fromBranch(branch);
+	$: setContextStore(Ownership, selectedOwnership);
 	$: selected = setSelected($selectedFiles, branch);
 
+	$: setContextStore(Branch, branch);
+	$: setContextStore(
+		LOCAL_COMMITS,
```

Comment:
I'm not loving the fact that we're still dealing with type-unsafe getters and setters.

Part of me is thinking about a public interface like:

```ts
// context.ts
function buildContextStore<T>() {
  const identifier = Symbol();
  const get = () => getContextStore<T>(identifier),
  const set = (value: T) => setContextStore(identifier, value)
  return [get, set]
}

// branchContexts.ts
const [getLocalCommitContextStore, setLocalCommitContextStore] = buildContextStore<LocalCommit[]>

export default { getLocalCommitContextStore, setLocalCommitContextStore }

// BranchLane.svelte
$: setLocalCommitContextStore(branch.commits.filter((c) => c.status == 'local'))

// Where its used
const localCommits = getLocalCommitContextStore()
```
would be nice.


I'm also not convinced about the symbols living in the types.ts file. I'm willing to be swayed on this, but it feels like it would make sense to have a file that is correlated with the component that creates the contexts which contains the relevant symbols

Reply:
That looks pretty great to me. Would you want to try and add something like that to this pr?

Reply:
Sure, I'm happy to have a go tomorrow 👍